### PR TITLE
Containerize Travis CI Builds

### PR DESCRIPTION
**Pull request by corranwebster**
In preparation for Python 3 stuff.

### CODE REVIEW

**Comment by corranwebster** [comment]
Well, that's not going to work since it calls out to `travis_setup.sh` which `sudo`'s a bunch of stuff.

**Comment by jwiggins** [comment]
The `sudo` commands are all for installing dependencies. If there was a clean way of installing everything in a venv, you could sidestep the need for root privs :wink:

**Comment by jwiggins** [comment]
Or... that. Fingers crossed!

**Comment by corranwebster** [comment]
As expected, PIL seems unhappy. :(

**Comment by corranwebster** [comment]
Why do we need to continue to support PIL again?  Would it be OK to just to CI with Pillow going forward?

**Comment by corranwebster** [comment]
Current state - pycairo is being painful, not sure about PIL.

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,26 @@
...
+    - python-wxtools
+    - ccache
+    - cmake
+    - swig
+    - zlib1g-dev
+    - libpng-dev
+    - libfreetype6-dev
+    - libcairo2-dev
+    - python-cairo
+    - python-cairo-dev
```
Comment (itziakos) [inline .travis.yml]: do we need the cairo dev?

File: dev_requirements_pil.txt [hunk 2]
```diff
@@ -1,8 +1,11 @@
 --allow-external PIL
 --allow-unverified PIL
+--allow-external pycairo
```
Comment (itziakos) [inline dev_requirements_pil.txt]: if we have pycairo then we do not need to install it here, unless we install a newer version correct?

**Comment by itziakos** [comment]
I would suggest that this PR waits for #208 before merging. I think it would be easier that way

**Comment by corranwebster** [comment]
I agree on #208, but would like to prove out the containerized building in the meantime.

**Comment by itziakos** [comment]
May I also suggest to start using codecov instead of coveralls? 
```
after_success:
  - pip install codecov
  - codecov
```

File: .travis.yml [hunk 3]
```diff
@@ -1,7 +1,26 @@
...
+    - python-pip
+    - python-numpy
+    - python-wxtools
+    - ccache
+    - cmake
+    - swig
+    - zlib1g-dev
+    - libpng-dev
+    - libfreetype6-dev
+    - libcairo2-dev
```
Comment (itziakos) [inline .travis.yml]: We should also not need to install this, we should let `python-cairo` bring any dependencies it needs.

**Comment by jwiggins** [comment]
#208 is merged

**Comment by corranwebster** [comment]
OK, I think that this is done.

**Comment by itziakos** [comment]
:+1:

**Comment by jwiggins** [comment]
:+1: Looks like a nice performance boost

**Pull request merged**